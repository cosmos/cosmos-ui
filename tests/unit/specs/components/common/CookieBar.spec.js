import { shallowMount } from "@vue/test-utils"
import CookieBar from "common/CookieBar"

describe(`CookieBar`, () => {
  let wrapper, $store

  beforeEach(async () => {
    $store = {
      commit: jest.fn(),
      getters: {
        session: {
          cookiesAccepted: false
        }
      }
    }

    wrapper = shallowMount(CookieBar, {
      mocks: {
        $store
      },
      stubs: [`router-link`]
    })
  })

  it(`shows the cookie bar`, () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it(`can triggers cookie acceptance on close`, () => {
    const dispatch = jest.fn()
    CookieBar.methods.accept.call({
      $store: {
        dispatch
      }
    })
    expect(dispatch).toHaveBeenCalledWith(`setAnalyticsCollection`, true)
    expect(dispatch).toHaveBeenCalledWith(`setErrorCollection`, true)
    expect(dispatch).toHaveBeenCalledWith(`storeLocalPreferences`)
  })
})
