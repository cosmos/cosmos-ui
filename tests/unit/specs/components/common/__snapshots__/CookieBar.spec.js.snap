// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookieBar shows the cookie bar 1`] = `
<div
  class="cookie-bar"
>
  <i />
   
  <p>
    <span
      class="hide-on-mobile"
    >
      This site uses cookies to help improve the user experience.
    </span>
    
    By using Lunie, you accept our
    
    <router-link-stub
      class="link"
      to="/terms"
    >
      Terms of Service
    </router-link-stub>
     and
    
    <router-link-stub
      class="link"
      to="/privacy"
    >
      Privacy Policy
    </router-link-stub>
    .
  
  </p>
   
  <a
    class="close"
  >
    <i
      class="material-icons"
    >
      close
    </i>
  </a>
</div>
`;
